### PR TITLE
[KYUUBI #1498] Support operation log for ExecuteScala

### PR DESCRIPTION
**Pull request by cxzl25**
### _Why are the changes needed?_
close https://github.com/apache/incubator-kyuubi/issues/1498

Pass scala repl output to operation log.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [x] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cxzl25** [comment]
## Test script
```sql
set kyuubi.operation.language=scala;
val df = spark.range(0, 10, 2, 1).toDF;
df.show();
result.set(df);
```

## Current
image

image

image


## Fix

image

image

image

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteScala.scala [hunk 1]
```diff
@@ -71,9 +75,12 @@ class ExecuteScala(
             if (result != null) {
               new ArrayFetchIterator[Row](result.collect())
             } else {
-              // TODO (#1498): Maybe we shall pass the output through operation log
-              // but some clients may not support operation log
-              new ArrayFetchIterator[Row](Array(Row(repl.getOutput)))
+              if (scalaReplToOperationLog) {
+                info("\n" + repl.getOutput)
+                new ArrayFetchIterator[Row](Array.empty[Row])
```
Comment (ulysses-you) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteScala.scala:80]: shall we also pass repl.output to client ?

**Comment by yaooqinn** [comment]
Maybe we do not need this config and print it to the operation log all the time, the output seems useless in result

**Comment by cxzl25** [comment]
> Maybe we do not need this config and print it to the operation log all the time, the output seems useless in result

I read the previous comment considering that some low-level clients do not support operation log, and query log is only supported in Hive 0.14 because of [HIVE-4629](https://issues.apache.org/jira/browse/HIVE-4629).

Or remove this configuration, as suggested by ulysses-you, output a copy of both operation log and output?

**Comment by ulysses-you** [comment]
if there is no affect for lower version client, I think we can just pass output to operation log and client

**Review by yaooqinn — APPROVED** [review]

**Comment by cxzl25** [comment]
I added the scala repl output prefix to make it easier to find.

##  Fix

image

image

image

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**